### PR TITLE
FgFlex: debug cnn dim and kernel size

### PR DESCRIPTION
**Pull request by pmhalvor**
After last PR merge, jobs were queued. Most kept running, but both `cnn_dim` and `kernel_size` failed due to mismatching sizes.

This PR cleans up these bugs and checks different setups in notebook.
These jobs need to be queued up again, as well as `attn`, since that one will inevitably fail after first run

### CODE REVIEW

File: src/model.py [hunk 1]
```diff
@@ -1985,33 +1985,33 @@ def init_components(self, subtasks):
                     ## CNN component: three possible cnn types for subtasks
                     if expanding_cnn:  # Sometimesse will be 0, other times None
                         components[task][f"cnn_{stack}"] = torch.nn.Sequential(*[
-                            self.expanding_cnn_block(cnn_dim*2, cnn_dim, kernel_size, m=expanding_cnn)
+                            self.expanding_cnn_block((768+cnn_dim), cnn_dim, kernel_size, m=expanding_cnn)
```
Comment (pmhalvor) [inline src/model.py:1988]: 768 since all tasks get original embeddings 
+
cnn dim, since this will be the output of shared task.

File: src/model.py [hunk 2]
```diff
@@ -2048,22 +2048,21 @@ def init_components(self, subtasks):
         relations = torch.nn.ModuleDict({})
         attn_linears = torch.nn.ModuleDict({})
         for stack in range(stack_count):
+            relations[f"stack_{stack}"] = torch.nn.ModuleDict({})
             for rel in attention_relations:
-                relations[f"stack_{stack}"] = torch.nn.ModuleDict({
-                    f"{rel[0]}_at_{rel[1]}": torch.nn.ModuleDict({
-                        "attn": self.attn_block(cnn_dim),
+                relations[f"stack_{stack}"][f"{rel[0]}_at_{rel[1]}"] = torch.nn.ModuleDict({
+                    "attn": self.attn_block(cnn_dim),
```
Comment (pmhalvor) [inline src/model.py:2054]: This was previously causing all attentions relations for a given stack except for the last to be overwritten. Now, the stack dict is created first, then relations are added as looped through

File: src/model.py [hunk 3]
```diff
@@ -2048,22 +2048,21 @@ def init_components(self, subtasks):
...
 
-                        # new linear for each attn relation for first_task logits using attn information
-                        "linear": self.linear_block(cnn_dim*2, 3),
-                    })
+                    # new linear for each attn relation for first_task logits using attn information
+                    "linear": self.linear_block(cnn_dim*2, 3)
```
Comment (pmhalvor) [inline src/model.py:2057]: Notice this input size remains the same. Due to attention output w/ cnn output (no embeddings here)

File: src/model.py [hunk 4]
```diff
@@ -2048,22 +2048,21 @@ def init_components(self, subtasks):
...
                 })
                 if rel[0] in self.subtasks + ["shared"] and "re_encode" not in components[rel[0]].keys():
                     # only reach this one time per first_task=rel[0]
                     count = sum([rel[0] == r[0] for r in attention_relations])
                     # map subtask information back to shared hidden state size after (multiple) attention(s)
                     components[rel[0]]["re_encode"] = self.linear_block(
                         in_features=cnn_dim*(1 + count),  # 1 task output + number of relations with rel[0] as first task
-                        out_features=cnn_dim*2
+                        out_features=(768+cnn_dim)
```
Comment (pmhalvor) [inline src/model.py:2065]: Final re-encoding also needs to be reshaped, with embedding size in mind

**Review by pmhalvor — COMMENTED** [review]
Most cnn input sizes needed to change, but not all. 

Also, pointed out where the change fixing attention relations occurs. 

See comments below
